src/main/java/com/google/api/gax/core/NanoClock.java

```diff
@@ -29,23 +29,33 @@
  */
 package com.google.api.gax.core;
 
-/**
- * An interface for getting the current value of a high-resolution time source, in nanoseconds.
- *
- * Clocks other than DefaultNanoClock are typically used only for testing.
- *
- * This interface is required in addition to Java 8's Clock, because nanoTime is required to compare
- * values with io.grpc.CallOptions.getDeadlineNanoTime().
- */
-public interface NanoClock {
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+/** Default implementation of the ApiClock interface, using call to System.nanoTime(). */
+public final class NanoClock implements ApiClock, Serializable {
```

Comment:
For SystemClock "system" is not about class name used in implementation, it is about actual system (machine) it is running on. NanoClock - just says that it has nano precision, and does not guarantee that the time is bounded to any real point in time.

Or please suggest 2 other names to those classes, but such that it is clear that SystemClock provides real world time, while NanoClock just a way to measure time difference between two events on same machine.